$OpenBSD: patch-configure,v 1.2 2014/09/22 15:15:41 jeremy Exp $

Fix so name, checking for DOT and DOXYGEN.

Override the arch setting to remove OpenBSD version from it,
so ports don't have to be bumped when OpenBSD version changes.

Remove invalid shell syntax, fixed upstream in r45715.

--- configure.orig	Thu Sep 18 10:04:39 2014
+++ configure	Fri Sep 19 09:12:21 2014
@@ -21056,7 +21056,7 @@ esac
   openbsd*|mirbsd*) :
 
 	SOLIBS='$(LIBS)'
-	LIBRUBY_SO='lib$(RUBY_SO_NAME).so.$(MAJOR).'`expr ${MINOR} \* 10 + ${TEENY}`
+	LIBRUBY_SO='lib$(RUBY_SO_NAME).so.'${LIBruby21_VERSION}
 	 ;; #(
   solaris*) :
 
@@ -21538,8 +21538,8 @@ if test x$setjmp_prefix = xsig; then
 else
     unset setjmp_sigmask
 fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: ${setjmp_prefix}setjmp${setjmp_suffix}${setjmp_cast:+($setjmp_cast)}" >&5
-$as_echo "${setjmp_prefix}setjmp${setjmp_suffix}${setjmp_cast:+($setjmp_cast)}" >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: ${setjmp_prefix}setjmp${setjmp_suffix}" >&5
+$as_echo "${setjmp_prefix}setjmp${setjmp_suffix}" >&6; }
 cat >>confdefs.h <<_ACEOF
 #define RUBY_SETJMP(env) ${setjmp_prefix}setjmp${setjmp_suffix}($setjmp_cast(env)${setjmp_sigmask+,0})
 _ACEOF
@@ -22300,7 +22300,10 @@ _ACEOF
 _ACEOF
 
 else
-    arch="${target_cpu}-${target_os}"
+    case "$target_os" in
+      openbsd*) arch="$target_cpu-openbsd" ;;
+      *) arch="${target_cpu}-${target_os}" ;;
+    esac
     cat >>confdefs.h <<_ACEOF
 #define RUBY_PLATFORM "${arch}"
 _ACEOF
